docs/streams/developer-guide.html

```diff
@@ -267,21 +295,532 @@
+    <table class="data-table" border="1">
+  <tbody><tr>
+          <th>Reading from Kafka</th>
+          <th>Description</th>
+  </tr>
+      <tr>
+        <td><b>Stream</b>: input topic(s) -> <code>KStream</code></td>
+        <td>Create a <code>KStream</code> from the specified Kafka input topic(s), interpreting the data as a record stream.
+	  A <code>KStream</code> represents a partitioned record stream. 
+	  <p>
+	  Slightly simplified, in the case of a KStream, the local KStream instance of every application instance will be populated
+	  with data from only a subset of the partitions of the input topic. Collectively, i.e. across all application instances,
+	  all the partitions of the input topic will be read and processed.
+	  <p>
+	  <pre class="brush: java;">
+	  import org.apache.kafka.common.serialization.Serdes;
+	  import org.apache.kafka.streams.kstream.KStreamBuilder;
+	  import org.apache.kafka.streams.kstream.KStream;
+
+	  KStreamBuilder builder = new KStreamBuilder();
+
+	  KStream&lt;String, Long&gt; wordCounts = builder.stream(
+	  Serdes.String(), /* key serde */
+	  Serdes.Long(),   /* value serde */
+	      "word-counts-input-topic" /* input topic */);
+	      </pre>
+	  When to provide serdes explicitly:
+	  <ul>
+	  <li>If you do not specify serdes explicitly, the default serdes from the configuration are used.</li>
+	  <li>You must specificy serdes explicitly if the key and/or value types of the records in the Kafka input topic(s) do not match
+	  the configured default serdes. </li>
+	  Several variants of <code>stream</code> exist to e.g. specify a regex pattern for input topics to read from.</td>
+      </tr>
+      <tr>
+        <td><b>Table</b>: input topic(s) -> <code>KTable</code></td>
+	<td>
+	  Reads the specified Kafka input topic into a <code>KTable</code>. The topic is interpreted as a changelog stream,
+	  where records with the same key are interpreted as UPSERT aka INSERT/UPDATE (when the record value is not <code>null</code>) or
+	  as DELETE (when the value is null) for that key. 
+	  <p>
+	  Slightly simplified, in the case of a KTable, the local KTable instance of every application instance will be populated
+	  with data from only a subset of the partitions of the input topic. Collectively, i.e. across all application instances, all
+	  the partitions of the input topic will be read and processed.
+	  <p>
+	  You must provide a name for the table (more precisely, for the internal state store that backs the table). This is required,
+	  among other things, for supporting interactive queries against the table.
+
+	  <pre class="brush: java;">
+        import org.apache.kafka.common.serialization.Serdes;
+        import org.apache.kafka.streams.kstream.KStreamBuilder;
+        import org.apache.kafka.streams.kstream.KTable;
+
+        KStreamBuilder builder = new KStreamBuilder();
+
+        KTable&lt;String, Long&gt; wordCounts = builder.table(
+	        Serdes.String(), /* key serde */
```

Comment:
The indentation looks wrong here -- just 1 blank for the Serdes.String() etc. lines.

Screenshot:

<img width="616" alt="screen shot 2017-07-17 at 17 05 58" src="https://user-images.githubusercontent.com/294849/28274785-3d90d992-6b12-11e7-925e-bc8523b1fcc3.png">